File: UNP/2023/2 qui/TAD/complexo.c
#include <stdio.h>
#include <stdlib.h>
// definição do tipo
typedef struct
{
    float a;
    float b;
} Complexo;
// Funcionalidades
Complexo *cria_complexo();
Complexo adiciona_complexo(Complexo n1, Complexo n2);
Complexo subtrair_complexo(Complexo n1, Complexo n2);
Complexo multiplica_complexo(Complexo n1, Complexo n2);
Complexo divide_complexo(Complexo n1, Complexo n2);
void exibe_complexo(Complexo n1);
Complexo conjugado(Complexo n1);

Complexo *cria_complexo()
{
    Complexo *temp = (Complexo *)malloc(sizeof(Complexo));
    return temp;
}

int main()
{
    Complexo n1;
    exibe_complexo(n1);
}

File: UNP/2023/2 qui/TAD/Complexo.java
public class Complexo {
    float a;
    float b;

    public Complexo(float n1, float n2) {
        a = n1;
        b = n2;
    }

    public void exibeComplexo(Complexo n1) {
        System.out.println("(" + a + " + " + b + " i)");
    }

}